packages/gafl-webapp-service/src/pages/recurring-payments/set-up-payment/__tests__/route.spec.js

```diff
@@ -0,0 +1,133 @@
+      const returnValue = Symbol('return value')
+      displayPermissionPrice.mockReturnValueOnce(returnValue)
+
+      const result = await getData(getSampleRequest())
+
+      expect(result.cost).toEqual(returnValue)
+    })
+
+    it('displayPermissionPrice is called with permission and getCatalog', async () => {
+      const catalog = Symbol('mock catalog')
+      const permission = Symbol('mock permission')
+      const request = getSampleRequest(permission, catalog)
+
+      await getData(request)
+
+      expect(displayPermissionPrice).toHaveBeenCalledWith(permission, catalog)
+    })
+
+    it('type equals return of recurringLicenceTypeDisplay', async () => {
+      const returnValue = Symbol('return value')
+      recurringLicenceTypeDisplay.mockReturnValueOnce(returnValue)
+
+      const result = await getData(getSampleRequest())
+
+      expect(result.type).toEqual(returnValue)
+    })
+
+    it('recurringLicenceTypeDisplay is called with permission and getCatalog', async () => {
+      const catalog = Symbol('mock catalog')
+      const permission = Symbol('mock permission')
+      const request = getSampleRequest(permission, catalog)
+
+      await getData(request)
+
+      expect(recurringLicenceTypeDisplay).toHaveBeenCalledWith(permission, catalog)
+    })
+
+    it('reminder equals return of recurringPayReminderDisplay', async () => {
+      const returnValue = Symbol('return value')
+      recurringPayReminderDisplay.mockReturnValueOnce(returnValue)
+
+      const result = await getData(getSampleRequest())
+
+      expect(result.reminder).toEqual(returnValue)
+    })
+
+    it('recurringPayReminderDisplay is called with permission and getCatalog', async () => {
+      const catalog = Symbol('mock catalog')
+      const permission = Symbol('mock permission')
+      const request = getSampleRequest(permission, catalog)
+
+      await getData(request)
+
+      expect(recurringPayReminderDisplay).toHaveBeenCalledWith(permission, catalog)
+    })
+
+    it('single equals return of addLanguageCodeToUri with choose payment', async () => {
+      const returnValue = Symbol('return value')
+      addLanguageCodeToUri.mockReturnValue(returnValue)
+
+      const result = await getData(getSampleRequest())
+
+      expect(result.uri.single).toEqual(returnValue)
+    })
+
+    it('terms equals return of addLanguageCodeToUri with terms and conditions', async () => {
+      const returnValue = Symbol('return value')
+      addLanguageCodeToUri.mockReturnValue(returnValue)
+
+      const result = await getData(getSampleRequest())
+
+      expect(result.uri.single).toEqual(returnValue)
+    })
+
+    it.each([[CHOOSE_PAYMENT.uri], [TERMS_AND_CONDITIONS.uri]
+    ])('addLanguageCodeToUri is called with permission and %s', async (uri) => {
+      const request = getSampleRequest()
+
+      await getData(request)
+
+      expect(addLanguageCodeToUri).toHaveBeenCalledWith(request, uri)
+    })
+  })
```

Comment:
All these tests are almost identical - please have a look at combining them all via it.each